src/bun.js/webcore/request.zig

```diff
@@ -59,8 +59,8 @@ pub const Request = struct {
     method: Method = Method.GET,
     request_context: JSC.API.AnyRequestContext = JSC.API.AnyRequestContext.Null,
     https: bool = false,
-    upgrader: ?*anyopaque = null,
-
+    finalized: bool = false,
```

Comment:
What makes it weak vs strong? The reference count is keeping the Request from being deinitialized

Couldn’t we do the same thing except clearer using bun.NewRefCounted?

Reply:
Weak values are not usable after being finalized and can be partially freed before (everything just not the main struct), so this uses less memory overall and avoid cycles refs for example, same technique is used by rust/c++. BunRefCounted is used for strong ref not weak the principle is different.